### PR TITLE
Generate live project documentation using mkdocs and gh-pages

### PR DESCRIPTION
**Pull request by aledbf**
*Note:* to build I need to remove `docs/apireference` directory. Can we move that directory?

**Local test:**
```
rm -rf docs/apireference
make live-docs

open browser localhost:3000
```

**Create a detached branch:**
```
git checkout --orphan gh-pages
git rm -rf .
touch .empty
git add .empty
git commit -m "Initial Commit"
```

After this step it is possible to build and publish the docs running:

```
GH_TOKEN=XXXXXX GH_REF=github.com/kubernetes/kops ./hack/publish-docs.sh
```

### CODE REVIEW

**Comment by aledbf** [comment]
@justinsb ping

**Comment by chrislovecnm** [comment]
> Note: to build I need to remove docs/apireference directory. Can we move that directory?

Yes we can move it, but we need to fix apireference already.  What is the purpose of this PR btw?  I know what it does, but in the big picture what value are we getting?

File: Makefile [hunk 1]
```diff
@@ -711,3 +711,14 @@ bazel-version-dist: bazel-crossbuild-nodeup bazel-crossbuild-kops bazel-protokub
 .PHONY: bazel-upload
 bazel-upload: bazel-version-dist # Upload kops to S3
 	aws s3 sync --acl public-read ${BAZELUPLOAD}/ ${S3_BUCKET}
+
+#-----------------------------------------------------------  
+# static html documentation  
+
+.PHONY: live-docs
+live-docs:
+	@docker run --rm -it -p 3000:3000 -v ${PWD}:/docs aledbf/mkdocs:0.1
```
Comment (chrislovecnm) [inline Makefile]: > aledbf/mkdocs:0.1

I am not a huge fan of pulling in external containers. I know you are maintaining this, but any other options?

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Kubernetes release.  kops does not stop a user from installing mismatching
+versions of K8s, but Kubernetes releases always require kops to install specific
+versions of components like docker, that tested against the particular
+Kubernetes version.
+
+#### Compatibility Matrix
+
+| kops version | k8s 1.5.x | k8s 1.6.x | k8s 1.7.x | k8s 1.8.x | k8s 1.9.x |
+|--------------|-----------|-----------|-----------|-----------|-----------|
+| 1.9.x        | Y         | Y         | Y         | Y         | Y         |
```
Comment (chrislovecnm) [inline docs/index.md]: We need 1.10 included

File: hack/publish-docs.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+git clone --branch=gh-pages --depth=1 https://${GH_REF} ${DIR}/gh-pages
+cd ${DIR}/gh-pages
+
+git rm -r .
+
+cp -r ${DIR}/../site/* .
+
+git add .
+git commit -m "Deploy GitHub Pages"
+git push --force --quiet "https://${GH_TOKEN}@${GH_REF}" gh-pages > /dev/null 2>&1
```
Comment (chrislovecnm) [inline hack/publish-docs.sh]: Why are we looking at gh-pages?  We really need our docs on kubernetes.io :shrug

**Comment by aledbf** [comment]
> I like this, and appreciate the work, but our goal has been to get our docs on kubernetes.io.

Who's working on that?

> but our goal has been to get our docs on kubernetes.io.

I think the goal should be provide a better UX for user documentation, including search.

> I do not think other k8s projects are maintaining gh-pages. Thoughts?

I already do this https://kubernetes.github.io/ingress-nginx/
This was the principal complaint from users  (from the survey https://docs.google.com/forms/d/e/1FAIpQLSdcqdfMbuJHeQQe6hB7iU7sf3q9qZlrrXtrwGPlS7sicOpICA/viewform)
Since the publication of the site, we have more metrics about that they search, how they browse the docs and most importantly we now double the number of contributions improving documentation.

**Comment by aledbf** [comment]
@chrislovecnm I can close this PR if the goal is to get the docs on kubernetes.io

**Comment by justinsb** [comment]
This is awesome, thank you @aledbf .  We'll have to reorg our docs a little, but I think people want to do that anyway.  I'm going to go ahead and merge this so we can get the reorg started :-)

**Pull request merged**